### PR TITLE
Auto-delete on app uninstall

### PR DESCRIPTION
**Pull request by jacquarg**
Move download dir to android external cache dir.

### CODE REVIEW

File: www-src/app/replicator/init.coffee [hunk 1]
```diff
@@ -829,6 +845,8 @@ module.exports = class Init
     # - mConfig
     # - mRemoteRequest
     migrations:
+        '0.2.1': # Move cache root directory to external storage cache
+            states: [ 'mMoveCache']
         '0.2.0':
             # Filters: upper version of platform requiered.
             states: ['mLocalDesignDocuments', 'mCheckPlatformVersions', \
```
Comment (kosssi) [inline www-src/app/replicator/init.coffee]: it's possible to delete all ": states: []" ?

**Comment by kosssi** [comment]
it's possible to create a folder app/replicator/migration and create a file with 0.2.1.coffee with all need in ? the same for all migration ?

File: www-src/app/replicator/init.coffee (hunk 1 above)
Comment (jacquarg) [inline www-src/app/replicator/init.coffee]: I am not sure right now, but I will check (and tweak) it, it will be safer.

**Comment by jacquarg** [comment]
I though about a solution like that. But keep in mind that the file would not contain all the needs for a specific migration as usually (not in the 0.2.1 case OK ) migration use standard steps (putRemoteRequest, checkPermission) which we shouldn't duplicate

**Pull request merged**